### PR TITLE
World: handle name collisions like Model

### PR DESCRIPTION
**Pull request by scpeters**
# 🦟 Bug fix

Follow-up to https://github.com/gazebosim/sdformat/pull/1293#discussion_r1287959178

## Checklist
- [X] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: src/World.cc [hunk 1]
```diff
@@ -283,21 +285,35 @@ Errors World::Load(sdf::ElementPtr _sdf, const ParserConfig &_config)
...
+        int i = 0;
+        while (frameNames.count(frameName) > 0)
+        {
+          frameName = frame.Name() + "_frame" + std::to_string(i++);
+        }
+        std::stringstream ss;
+        ss << "Frame with name [" << frame.Name() << "] "
+           << "in world with name [" << this->Name() << "] "
+           << "has a name collision, changing frame name to ["
+           << frameName << "].";
```
Comment (scpeters) [inline src/World.cc:302]: related to https://github.com/gazebosim/sdformat/issues/1312

I can revert this if desired

**Comment by scpeters** [comment]
this is a behavior change, so we could consider targeting `main`

**Review by azeey — APPROVED** [review]

**Comment by azeey** [comment]
> this is a behavior change, so we could consider targeting `main`

It's probably best if we target `main`

**Comment by scpeters** [comment]
> > this is a behavior change, so we could consider targeting `main`
> 
> It's probably best if we target `main`

I rebased on `main` and updated the Migration guide in https://github.com/gazebosim/sdformat/pull/1311/commits/75bb46533cc4f88c40d4278d7168f9856222ae5c

**Review by azeey — APPROVED** [review]

**Pull request merged**